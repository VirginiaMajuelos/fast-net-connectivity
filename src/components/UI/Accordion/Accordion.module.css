.accordion {
  border-radius: 15px;
  background: var(--color-blue-50);
  color: var(--basic-dark);
  margin: 1rem;
  box-shadow: 0px 2px 8px rgba(0, 0, 0, 0.1607843137);
}

.accordion__header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0.75rem;
  cursor: pointer;
  border-radius: 15px;
  /* border: 1px solid var(--color-blue-400); */
}

.accordion__title {
  font-size: 1.125rem;
  font-weight:var(--font-weight-regular, 600);
  margin: 3px;
}

.accordion__icon {
  font-size: 1.5rem;
  margin: 3px;
  transition: transform 0.5s ease-in-out;
}

.accordion__icon:hover {
  color: var(--secondary);
}

.accordion__icon__rotate {
  transform: rotate(180deg);
}

.accordion__body {
  height: 0;
  opacity: 0;
  transition: opacity 0.5s ease-in-out;
  overflow: hidden;
  border-radius: 15px;
  /* border: 1px solid var(--color-blue-400);
  border-top: none; */
}

.accordion__body__show {
  height: auto;
  opacity: 1;
  padding: 1rem;
}
