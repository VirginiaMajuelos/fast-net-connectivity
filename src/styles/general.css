h1{
  font-size: var(--font-size-h1);
  font-weight: var(--font-weight-bold);
  font-family: var(--font-family-secondary);
}

h2 {
  font-size: var(--font-size-h2);
  font-weight: var(--font-weight-bold);
  font-family: var(--font-family-secondary);
}

h3 {
  font-size: var(--font-size-h3);
  font-weight: var(--font-weight-bold);
  font-family: var(--font-family-secondary);
  color: var(--color-blue-800)
}

h4{
  font-size: var(--font-size-h4);
  font-weight: var(--font-weight-bold);
  font-family: var(--font-family-secondary);
}

p {
  font-size: var(--font-size-p);
  font-weight: var(--font-weight-regular);
  font-family: var(--font-family-primary);
  line-height: var(--font-height-regular);
}

section {
  padding: var(--spacing-lg, 32px);
  margin-bottom: 40px;
}

@media (max-width: 1024px) {
  section {
    padding: var(--spacing-md, 24px);
    margin-bottom: 32px;
  }
}

@media (max-width: 768px) {
  section {
    padding: var(--spacing-md, 20px) var(--spacing-sm);
    margin-bottom: 24px;
  }
}

.mb0 {
  margin-bottom: var(--margin-bottom);
}

.container__box {
  display: grid;
  gap: 1rem;
  grid-template-columns: repeat(2, 1fr); 
  height: auto;
  margin-top: 1rem;
}

.container__box--2column {
  grid-template-columns: repeat(2, 1fr);
}

.container__title {
  margin: 30px;
}

.container__title h3 {
  font-size: var(--font-size-xl);
}

.container__name {
  display: flex;
  flex-direction: row;
  gap: 0.5rem;
  justify-content: center;
}

.box {
  height: 100%;
  margin: auto;
  border-radius: 5px;
  background-color: #7b7b7b1f;
  padding: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.box__isActive {
  border: 1px solid var(--color-blue-300);
}

.box__fibra {
  height: 100%;
  width: fit-content;
  margin: auto;
  border-radius: 5px;
  background-color: #7b7b7b1f;
  padding: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.productImage {
  width: 100px;
  height: 70px;
}

.container__list {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  gap: 0.5rem;
  align-items: baseline;
}

.description {
  text-align: center;
  font-weight: 300;
  margin-bottom: 0;
}

.title__card {
  text-align: center;
  font-size: 20px;
  display: flex;
  gap: 1rem;
  align-items: center;
}

.description__list {
  text-align: center;
  margin-top: 1rem;
  margin-bottom: 0;
  text-align:left;

}

.price {
  font-size: 3rem;
  font-weight: 600;
}

@media (max-width: 768px) {
  .container__box {
    grid-template-columns: repeat(2, 1fr); 
  }

  .container__box--2column {
    grid-template-columns: repeat(1, 1fr); 
  }

  .container__title h3 {
    font-size: var(--font-size-lg); 
  }

  .box {
    padding: 15px; 
  }

  .productImage {
    width: 80px; 
    height: 60px;
  }

  .price {
    font-size: 2.5rem; 
  }

  .container__list {
    flex-direction: row; 
    align-items: baseline; 
  }
}

@media (max-width: 480px) {
  .container__box {
    grid-template-columns: 1fr; 
  }

  .box {
    padding: 10px; 
  }

  .container__title h3 {
    font-size: var(--font-size-md); 
  }

  .price {
    font-size: 2rem; 
  }
}

.container__picker {
  display: flex;
  flex-direction: row;
  gap: 1rem;
  margin: 1rem 0 1rem 0;
  align-items: center;
}

.btn__picker {
  padding: 6px 14px;
  border: 1px solid var(--color-blue-300);
  background-color: var(--color-blue-50);
  cursor: pointer;
  border-radius: 5px;
}

.btn__pickerSelect {
  padding: 6px 14px;
  border: 2px solid var(--color-blue-700);
  background-color: var(--color-blue-100);
  border-radius: 5px;
  
}
