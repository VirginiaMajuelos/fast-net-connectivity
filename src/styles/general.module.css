.container__box {
  display: grid;
  gap: 1rem;
  grid-template-columns: repeat(2, 1fr); 
  height: auto;
  margin-top: 1rem;
}

.container__box--2column {
  grid-template-columns: repeat(2, 1fr);
}

.container__title {
  margin: 30px;
}

.container__title h3 {
  font-size: var(--font-size-xl);
}

.container__name {
  display: flex;
  flex-direction: row;
  gap: 0.5rem;
  justify-content: center;
}

.box {
  height: 100%;
  margin: auto;
  border-radius: 5px;
  background-color: #7b7b7b1f;
  padding: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.box__isActive {
  border: 1px solid var(--color-blue-300);
}

.box__fibra {
  height: 100%;
  width: fit-content;
  margin: auto;
  border-radius: 5px;
  background-color: #7b7b7b1f;
  padding: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.title__accordion {
  font-size: 2.2rem;
  font-weight: bold;
  color: var(--color-blue-800);
  padding-bottom: 1rem;
}

.productImage {
  width: 100px;
  height: 70px;
}

.container__list {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  gap: 0.5rem;
  align-items: baseline;
}

.description {
  text-align: center;
  font-weight: 300;
  margin-bottom: 0;
}

.title__card {
  text-align: center;
  font-size: 20px;
}

.description__list {
  text-align: center;
  margin-top: 1rem;
  margin-bottom: 0;
  /* font-size: 14px; */
  text-align:left;

}

.price {
  font-size: 3rem;
  font-weight: 600;
}

/* Responsive Styles */
@media (max-width: 768px) {
  .container__box {
    grid-template-columns: repeat(2, 1fr); /* 2 columns in tablet/mobile */
  }

  .container__box--2column {
    grid-template-columns: repeat(1, 1fr); /* 1 column on smaller screens */
  }

  .container__title h3 {
    font-size: var(--font-size-lg); /* Adjust title font size for smaller screens */
  }

  .box {
    padding: 15px; /* Reduce padding on smaller screens */
  }

  .productImage {
    width: 80px; /* Adjust image size */
    height: 60px;
  }

  .description {
    font-size: 0.9rem; /* Adjust description text size for better readability */
  }

  .price {
    font-size: 2.5rem; /* Adjust price size */
  }

  .container__list {
    flex-direction: column; /* Stack list items on smaller screens */
    align-items: center; /* Center the list items */
  }
}

@media (max-width: 480px) {
  .container__box {
    grid-template-columns: 1fr; /* 1 column layout for mobile */
  }

  .box {
    padding: 10px; /* Reduce padding even more on very small screens */
  }

  .container__title h3 {
    font-size: var(--font-size-md); /* Further reduce title font size */
  }

  .price {
    font-size: 2rem; /* Further adjust price font size */
  }
}


.container__picker {
  display: flex;
  flex-direction: row;
  gap: 1rem;
  margin: 1rem;
  align-items: center;
}

.btn__picker {
  padding: 6px 14px;
  border: 1px solid var(--color-blue-300);
  background-color: var(--color-blue-50);
  cursor: pointer;
  border-radius: 5px;
}

.btn__pickerSelect {
  padding: 6px 14px;
  border: 2px solid var(--color-blue-700);
  background-color: var(--color-blue-100);
  border-radius: 5px;
  
}
