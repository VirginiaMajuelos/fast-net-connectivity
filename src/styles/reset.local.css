@import './tokens.css';

/* Restablecer los márgenes y paddings predeterminados en todos los elementos */
*,
::before,
::after {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

/* Tipografía global */
html {
  font-family: var(--font-family-secondary);
  font-size: var(--font-size-md, 16px);
  line-height: 1.5;
}

/* Cuerpo de la página */
body {
  height: 100%;
  width: 100%;
  min-height: 100vh;
  line-height: 1.5;
  font-weight: var(--font-weight-regular, 400);
  margin: 0;
  transition: background-color 0.3s ease, color 0.3s ease;
  text-align: center;
  background-color: var(--color-blue-50);
}

/* Estilos básicos para títulos */
h1, h2, h3, h4, h5, h6 {
  font-weight: var(--font-weight-regular, 600);
  margin-bottom: var(--spacing-sm, 8px);
}

/* Estilos para párrafos */
p {
  margin-bottom: var(--spacing-md, 16px);
}

/* Espaciado entre secciones */
section {
  padding: var(--spacing-lg, 32px);
}

/* Enlaces */
a {
  color: inherit;
  text-decoration: none;
}

/* Reseteo de algunos estilos para listas */
ul, ol {
  list-style-type: none;
}

button {
  background: none;
  border: none;
  cursor: pointer;
}

input, button, textarea {
  font: inherit;
}

/* Clases para el tema Light */


/* Aplicación global de fondo y color de texto */
body main {
  background-color: var(--color-blue-100);
  color: var(--text-color);
}

h1{
  font-size: 36px;
  font-weight:400;
}
