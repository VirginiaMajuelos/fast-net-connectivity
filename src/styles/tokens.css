

:root {
--color-blue-50:  #ebf4ff;
--color-blue-100: #d6e8ff;
--color-blue-200: #b3d4ff;
--color-blue-300: #8abaff;
--color-blue-400: #67a2ff;
--color-blue-500: #7AAFF5; /* Color base */
--color-blue-600: #3e7ffe;
--color-blue-700: #2c5bcc;
--color-blue-800: #1d3e8a;
--color-blue-900: #0d1f45;
--color-blue-950: #071022;

--color-light-1: #f7f7f7;
--color-light-2: #dcdcdc;
--color-light-3: #aaaaaa;
--color-light-4: #484848;

--color-dark-1: #0c0a09;
--color-dark-2: #1c1c1c;
--color-dark-3: #3d3d3d;
--color-dark-4: #b6b6b6;

  /* Colores de texto para el modo Dark */
  --color-primary-dark: var(--color-yellow-600);

  /* Colores para el logo */
  --logo-color-light: var(--color-yellow-600);
  --logo-color-dark: var(--color-purple-700);

  /* Variables para los modos de color */
  --background-color-light: var(--color-purple-50);
  --text-color-light: var(--color-light-3);
  --background-color-dark: var(--color-purple-950);
  --text-color-dark: var(--color-primary-dark);

  /* Variables de tipografía */
  --font-family-primary: 'Arial', sans-serif;
  --font-family-secondary: 'Kanit', sans-serif;
  --font-weight-regular: 400;
  --font-weight-bold: 700;
  --font-height-regular: 1.6;

  /* Espaciado */
  --spacing-xs: 4px;
  --spacing-sm: 8px;
  --spacing-md: 16px;
  --spacing-lg: 32px;
  --spacing-xl: 64px;
  --margin-bottom: 0;
  
  /* Otros */
  --border-radius: 8px;
  --box-shadow-light: 0 2px 4px rgba(0, 0, 0, 0.1);
  --box-shadow-dark: 0 2px 4px rgba(0, 0, 0, 0.3);

  /* Tamaños de fuente base */
  --font-size-xs: 12px;
  --font-size-sm: 14px;
  --font-size-md: 16px;
  --font-size-lg: 18px;
  --font-size-xl: 24px;
  --font-size-xxl: 32px;

  /* Tamaños responsivos */
  --font-size-h1: 3rem;   /* 48px */
  --font-size-h2: 2.5rem; /* 40px */
  --font-size-h3: 2rem;   /* 32px */
  --font-size-h4: 1.5rem; /* 24px */
  --font-size-p: 1.125rem;  /* 18px */
}
/* Para Tablet */
@media (max-width: 1024px) { 
  :root {
    --font-size-h1: 2.5rem; /* 40px */
    --font-size-h2: 2rem;   /* 32px */
    --font-size-h3: 1.75rem; /* 28px */
    --font-size-h4: 1.25rem; /* 20px */
    --font-size-p: 1rem;    /* 18px */
  }
}

/* Para Mobile */
@media (max-width: 480px) { 
  :root {
    --font-size-h1: 1.75rem;  /* 32px */
    --font-size-h2: 1.75rem; /* 28px */
    --font-size-h3: 1.5rem;  /* 24px */
    --font-size-h4: 1.125rem; /* 18px */
  }
}

