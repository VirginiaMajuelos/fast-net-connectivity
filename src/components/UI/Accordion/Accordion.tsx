import React, { useState } from 'react';
import { AiOutlineDown } from 'react-icons/ai';

import styles from './Accordion.module.css';

interface AccordionProps {
  title: string;
  children: React.ReactNode;
}

export const Accordion: React.FC<AccordionProps> = ({ title, children }) => {
  const [show, setShow] = useState(false);

  const handleToggle = () => {
    setShow((prev) => !prev);
  };

  return (
    <div className={styles.accordion}>
      <div className={styles.accordion__header} onClick={handleToggle}>
        <p className={styles.accordion__title}>{title}</p>
        <AiOutlineDown
          className={`${styles.accordion__icon} ${
            show ? styles.accordion__icon__rotate : ''
          }`}
        />
      </div>
      <div
        className={`${styles.accordion__body} ${
          show ? styles.accordion__body__show : ''
        }`}
      >
        <p>{children}</p>
      </div>
    </div>
  );
};
