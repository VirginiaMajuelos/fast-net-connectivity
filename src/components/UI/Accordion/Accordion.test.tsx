import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { describe, expect, it } from 'vitest';

import { Accordion } from './Accordion';
import styles from './Accordion.module.css';


describe('Accordion component', ()=> {
  it('render component', () => {
    render(<Accordion title='titulo'>text</Accordion>);
    expect(screen.getByText('text')).toBeInTheDocument();
  });
  it('render title as a prop', () => {
    render(<Accordion title='titulo'>text</Accordion>);
    expect(screen.getByText('titulo')).toBeInTheDocument();
  });

  it('the content is hidden by default', () => {
    render(<Accordion title='string'>text</Accordion>);

    const content = screen.getByText('text');
    const contentWrapper= content.parentElement;
    expect(contentWrapper).toHaveClass(styles.accordion__body);
    expect(contentWrapper).not.toHaveClass(styles.accordion__body__show);
  });

  it('the content is showed when you click', async () => {
    const user = userEvent.setup();

    render(<Accordion title='My Title'>My Content</Accordion>);

    const header = screen.getByText('My Title');
    const content = screen.getByText('My Content');
    const wrapper = content.parentElement;

    //oculto por defecto
    expect(wrapper).not.toHaveClass(styles.accordion__body__show);

    //primer click: mostrar
    await user.click(header);
    expect(wrapper).toHaveClass(styles.accordion__body__show);

    //segundo click: ocultar
    await user.click(header);
    expect(wrapper).not.toHaveClass(styles.accordion__body__show);

  });

  it('Rotate icon change when the accordion is open', async () => {
    const user = userEvent.setup();
    render(<Accordion title='My Title'>My Content</Accordion>);
    
    const header = screen.getByText('My Title');
    const iconBtn = screen.getByLabelText('outline-down-icon');

    expect(iconBtn).not.toHaveClass(styles.accordion__icon__rotate);
    
    await user.click(header);
    expect(iconBtn).toHaveClass(styles.accordion__icon__rotate);

  });

  it('matches the snapshot', () => {
    const { asFragment } = render(
      <Accordion title='Snapshot Test'>Content snapshot</Accordion>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('has aria-label on the icon', () => {
    render(<Accordion title='Acc title'>Acc content</Accordion>);
    const icon = screen.getByLabelText('outline-down-icon');
    expect(icon).toBeInTheDocument();
  });

});