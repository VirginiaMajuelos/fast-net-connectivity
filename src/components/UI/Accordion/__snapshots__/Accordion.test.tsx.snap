// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Accordion component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="_accordion_224b33"
  >
    <div
      class="_accordion__header_224b33"
    >
      <p
        class="_accordion__title_224b33"
      >
        Snapshot Test
      </p>
      <svg
        aria-label="outline-down-icon"
        class="_accordion__icon_224b33 "
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
        />
      </svg>
    </div>
    <div
      class="_accordion__body_224b33 "
    >
      <p>
        Content snapshot
      </p>
    </div>
  </div>
</DocumentFragment>
`;
